assets/app/scripts/services/applicationGenerator.js

```diff
@@ -205,16 +205,7 @@ angular.module("openshiftConsole")
         });
       }
 
-      // User can input a URL that contains a ref
-      var uri = new URI(input.buildConfig.sourceUrl);
-      var sourceRef = uri.fragment();
-      if (!sourceRef || sourceRef.length === 0) {
```

Comment:
@jwforres Any reason the `sourceRef.length` check was needed here? Does `fragment()` not return a string?